spec/praxis/api_definition_spec.rb

```diff
@@ -0,0 +1,159 @@
+require 'spec_helper'
+
+describe Praxis::ApiDefinition do
+  
+  subject(:api){  Praxis::ApiDefinition.instance }  
+  
+  # Without getting a fresh new ApiDefinition it is very difficult to test stuff using the Singleton
+  # So for some tests we're gonna create a new instance and work with it to avoid the singleton issues
+  let(:non_singleton_api) do
```

Comment:
Sure. Something to keep in mind for later perhaps, but not a blocker by any means.